src/runtime/link-accounts-flow-test.js

```diff
@@ -470,4 +440,117 @@ describes.realWin('LinkSaveFlow', {}, env => {
       expect(result).to.be.true;
     });
   });
+
+  it('should should fail if both token and authCode are present', () => {
+    const reqPromise = new Promise(resolve => {
+      resolve({token: 'test', authCode: 'test'});
+    });
+    linkSaveFlow = new LinkSaveFlow(runtime, () => reqPromise);
+    let messageCallback = undefined;
+    sandbox.stub(port, 'onMessage', cb => {
+      messageCallback = cb;
+    });
+    activitiesMock.expects('openIframe').returns(Promise.resolve(port));
+    return linkSaveFlow.start().then(() => {
+      messageCallback({
+        'getLinkingInfo': true,
+      });
+      return linkSaveFlow.requestPromise_;
+    }).then(() => {
+      throw new Error('must have failed');
+    }, reason => {
+      expect(() => {
```

Comment:
At least one of failure tests should check that the dialog is closed.

Reply:
done, discussed offline.